### PR TITLE
Memleak fix and assorted fs code cleanups

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/analyze/analyze.c [hunk 1]
```diff
@@ -1972,7 +1981,7 @@ static int dump_filesystems(int argc, char *argv[], void *userdata) {
                 if (k < 0) {
                         fputc('\n', stdout);
                         fflush(stdout);
-                        log_notice_errno(k, "# Not showing unlisted filesystems, couldn't retrieve kernel filesystem list: %m");
+                                log_notice_errno(k, "# Not showing unlisted filesystems, couldn't retrieve kernel filesystem list: %m");
```
Comment (bluca) [inline src/analyze/analyze.c]: what's with this indentation change?

**Review by bluca — APPROVED** [review]
lgtm just one nit

**Comment by poettering** [comment]
lgtm (can't comment on the zsh parts, though)

File: src/analyze/analyze.c (hunk 1 above)
Comment (keszybz) [reply src/analyze/analyze.c]: Initially I wanted to add `if (!arg_quiet)` here, but then I added an early exit above. Thanks for catching this.

**Comment by keszybz** [comment]
> lgtm (can't comment on the zsh parts, though)

I tested the zsh parts, I think they should be OK.

**Comment by yuwata** [comment]
```
In file included from ../src/basic/macro.h:457,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/analyze/analyze.c:14:
../src/analyze/analyze.c: In function ‘dump_syscall_filters’:
../src/basic/log.h:204:27: error: ‘k’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  204 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                           ^~~~~~~~~~~~
../src/analyze/analyze.c:1711:21: note: ‘k’ was declared here
 1711 |                 int k;
      |                     ^
cc1: all warnings being treated as errors
```

**Comment by keszybz** [comment]
build (gcc): 322/1232 test-repart                             TIMEOUT 30.01 s

**Comment by keszybz** [comment]
focal-amd64 timed out in upstream tests.

**Pull request merged**